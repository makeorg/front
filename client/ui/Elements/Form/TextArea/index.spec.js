import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { TextArea } from './index';

jest.mock('../Styled/Input', () => ({
  FakeFieldStyle: 'FakeFieldStyle',
}));

jest.mock('../Styled/TextArea', () => ({
  TextAreaIconStyle: 'TextAreaIconStyle',
  BasicTextAreaStyle: 'BasicTextAreaStyle',
  TextAreaCounterStyle: 'TextAreaCounterStyle',
}));

const name = 'bar';
const icon = 'SvgPencil';
const value = 'baz';
const label = 'qux';
const handleChange = () => {};

describe('TextArea', () => {
  it('must return the diff between snapshot when TextArea is required or optionnal', () => {
    const RequiredTextArea = renderer
      .create(
        <TextArea
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
          required
        />
      )
      .toJSON();
    const OptionnalTextArea = renderer
      .create(
        <TextArea
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
          required={false}
        />
      )
      .toJSON();
    expect(snapshotDiff(RequiredTextArea, OptionnalTextArea)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when TextArea has a limit of characters or not', () => {
    const NoLimitTextArea = renderer
      .create(
        <TextArea
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
        />
      )
      .toJSON();
    const LimitedTextArea = renderer
      .create(
        <TextArea
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
          minLength="25"
          maxLength="450"
        />
      )
      .toJSON();
    expect(snapshotDiff(NoLimitTextArea, LimitedTextArea)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when TextArea has a enable vs disable spellchecking', () => {
    const EnabledSpellchecking = renderer
      .create(
        <TextArea
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
        />
      )
      .toJSON();
    const DisabledSpellchecking = renderer
      .create(
        <TextArea
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
          spellCheck={false}
        />
      )
      .toJSON();
    expect(
      snapshotDiff(EnabledSpellchecking, DisabledSpellchecking)
    ).toMatchSnapshot();
  });

  it('must return the diff between snapshot when TextArea has a disable vs enable autocomplete', () => {
    const DisabledAutocomplete = renderer
      .create(
        <TextArea
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
        />
      )
      .toJSON();
    const EnabledAutocomplete = renderer
      .create(
        <TextArea
          name={name}
          icon={icon}
          value={value}
          label={label}
          handleChange={handleChange}
          autoComplete="on"
        />
      )
      .toJSON();
    expect(
      snapshotDiff(DisabledAutocomplete, EnabledAutocomplete)
    ).toMatchSnapshot();
  });
});
