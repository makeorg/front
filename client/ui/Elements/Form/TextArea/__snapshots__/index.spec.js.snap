// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea must return the diff between snapshot when TextArea has a disable vs enable autocomplete 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -9,11 +9,11 @@
    </span>
    <div
      className="sc-1ln9y8s-4 bEGCGe"
    >
      <BasicTextAreaStyle
-       autoComplete="off"
+       autoComplete="on"
        id="bar"
        maxRows={25}
        name="bar"
        onChange={[Function anonymous]}
        required={false}
`;

exports[`TextArea must return the diff between snapshot when TextArea has a enable vs disable spellchecking 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -16,11 +16,11 @@
        maxRows={25}
        name="bar"
        onChange={[Function anonymous]}
        required={false}
        rows={5}
-       spellCheck={true}
+       spellCheck={false}
        value="baz"
      />
      <label
        className="sc-1ln9y8s-9 kZrAJF"
        htmlFor="bar"
`;

exports[`TextArea must return the diff between snapshot when TextArea has a limit of characters or not 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -11,11 +11,13 @@
      className="sc-1ln9y8s-4 bEGCGe"
    >
      <BasicTextAreaStyle
        autoComplete="off"
        id="bar"
+       maxLength="450"
        maxRows={25}
+       minLength="25"
        name="bar"
        onChange={[Function anonymous]}
        required={false}
        rows={5}
        spellCheck={true}
@@ -25,7 +27,10 @@
        className="sc-1ln9y8s-9 kZrAJF"
        htmlFor="bar"
      >
        qux
      </label>
+     <TextAreaCounterStyle>
+       3/450 common.max_characters
+     </TextAreaCounterStyle>
    </div>
  </div>
`;

exports[`TextArea must return the diff between snapshot when TextArea is required or optionnal 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -14,11 +14,11 @@
        autoComplete="off"
        id="bar"
        maxRows={25}
        name="bar"
        onChange={[Function anonymous]}
-       required={true}
+       required={false}
        rows={5}
        spellCheck={true}
        value="baz"
      />
      <label
`;
